### PR TITLE
Feature/yasin developments

### PR DESCRIPTION
**Pull request by YAS-SIIN**

### CODE REVIEW

File: tests/ZiraLink.Api.Application.UnitTests/Services/CustomerServiceTest.cs [hunk 1]
```diff
@@ -20,19 +30,122 @@ public CustomerServiceTest()
...
         [Theory]
-        [InlineData("4")]
+        [InlineData("1")]
         public async Task GetCustomerByExternalId_WhenEverythingIsOk_ShouldHasData(string externalId)
         {
             //Arrange
             var mockConfiguration = new Mock<IConfiguration>();
-            mockConfiguration.Setup(m => m["ZIRALINK_URL_IDS"]).Returns("https://ids.ziralink.local:5001");
-   
+            mockConfiguration.Setup(m => m[It.IsAny<string>()]).Returns("https://ids.ziralink.local:5001");
```
Comment (saeedmaghdam) [inline tests/ZiraLink.Api.Application.UnitTests/Services/CustomerServiceTest.cs:38]: `ZIRALINK_URL_IDS` should be remained, `It.IsAny<string>()` is not correct

File: tests/ZiraLink.Api.Application.UnitTests/Services/CustomerServiceTest.cs [hunk 2]
```diff
@@ -1,11 +1,21 @@
 ﻿
+using Castle.Core.Resource;
```
Comment (saeedmaghdam) [inline tests/ZiraLink.Api.Application.UnitTests/Services/CustomerServiceTest.cs:2]: suggest to remove all extra new lines from usings

**Review by saeedmaghdam — CHANGES_REQUESTED** [review]

File: tests/ZiraLink.Api.Application.UnitTests/Services/CustomerServiceTest.cs (hunk 1 above)
Comment (YAS-SIIN) [reply tests/ZiraLink.Api.Application.UnitTests/Services/CustomerServiceTest.cs:38]: Done.

File: tests/ZiraLink.Api.Application.UnitTests/Services/CustomerServiceTest.cs (hunk 2 above)
Comment (YAS-SIIN) [reply tests/ZiraLink.Api.Application.UnitTests/Services/CustomerServiceTest.cs:2]: Removed.

**Review by saeedmaghdam — APPROVED** [review]

**Pull request merged**